src/starplot/styles/helpers.py

```diff
@@ -38,9 +41,31 @@ def wrapper(*args, **kwargs):
                 else:
                     kwargs["style"] = style_class(**style)
 
-            elif style is None and style_attr is not None:
-                # if no style overrides and there's a base style, then just pass the base style
-                kwargs["style"] = getattr(args[0].style, style_attr, None)
+            elif style is None:
```

Comment:
There's few issues with this `elif`, but first let's take a step back and consider that there are three possible conditions we wanna check here:

1. Style overrides passed as a dict
2. Style overrides passed as kwargs (style__*)
3. No style overrides

We have 1 above covered by the first if block, and this second if block should cover condition 2 but it's not clear in the code that it should. To make it more clear, I think we should change this `elif style is None` to `elif style_kwargs`

The next issue is that on line 64, `styling_overrides` is undefined. I think we could fix that with a simple re-arranging though:

```python
if style and isinstance(style, dict): # this is condition 1
    ... 
elif style_kwargs: # this is condition 2
    # if style kwargs are used - build style dict and pass as "style" variable
    styling_overrides = {}
    for key, value in style_kwargs.items():
        parts = key.split("__")[1:]
        temp = styling_overrides
        for part in parts[:-1]:
            temp.setdefault(part, {})
            temp = temp[part]
        temp[parts[-1]] = value
        
    if style_attr is not None:
        base_style = getattr(args[0].style, style_attr).model_dump_json()
        base_style = json.loads(base_style)
        merge_dict(base_style, styling_overrides)
        kwargs["style"] = style_class(**base_style)
    else:
        kwargs["style"] = style_class(**styling_overrides)
```

This is getting closer, but then the next issue is with this:
```python
    styling_overrides = {}
    for key, value in style_kwargs.items():
        parts = key.split("__")[1:]
        temp = styling_overrides
        for part in parts[:-1]:
            temp.setdefault(part, {})
            temp = temp[part]
        temp[parts[-1]] = value
```
You can think of this block's purpose as just creating the dict `styling_overrides` properly (e.g. `{"marker": {"size": 20}}`) -- and I think it works for styles that have two layers, but not if they just have one layer (e.g. PolygonStyle which is used for plotting rectangles, circles, polygons). Something like [this](https://stackoverflow.com/a/35508197) should work for all cases, which looks pretty similar to what you have here but a few key differences.